src/components/Widgets/Form/AddressSelect.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types'
+import ReactSelect from 'react-select'
+import styled from 'styled-components'
+import Identicon from '../../Identicon'
+import { combineWallets } from '../../../util/helpers';
+
+export default class AddressSelect extends Component {
+  static displayName = 'AddressSelect';
+
+  static propTypes = {
+    disabled: PropTypes.bool,
+    wallets: PropTypes.object,
+    walletContracts: PropTypes.object,
+    onChange: PropTypes.func,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    wallets: {},
+    walletContracts: {},
+  };
+
+  state = {
+    selectedWallet: '',
+    combinedWallets: [],
+  }
+
+  componentDidMount() {
+    const { wallets, walletContracts, onChange } = this.props;
+    const combinedWallets = combineWallets(wallets, walletContracts);
+    let wallet = combinedWallets[0]
+    const address = wallet ? wallet.address : ''
+    this.setState(
+      { combinedWallets, selectedWallet: {value: address, ...wallet} },
+      onChange(combinedWallets[0])
+    )
+  }
+
+  selectAddress = e => {
+    const { onChange } = this.props;
+    this.setState({ selectedWallet: e }, onChange(e))
+  }
+
+  render() {
+    const { disabled } = this.props;
+    const { combinedWallets, selectedWallet } = this.state
+
+    let options = combinedWallets.map(w => {
+      return {
+        value: w.address,
+        label :w.addressType === 'wallet' 
+          ? `🔑 ${w.name}` + ` - ${w.balance} ETHER`
+          : w['contract-name'] + ` - ${w.balance} ETHER`,
```

Comment:
You're still mixing template literals with string concatenation here. The whole line can just be a template literal.